// tslib/subjects.js
export const ALL_SUBJECTS = [
    // --- NIVEAU 1 : INTRO (CHOIX DE DÉPART) ---
    {
        id: 'X01-A',
        name: 'Teach me something',
        videoUrl: '/assets/videos/X01-A.webm',
        timeline: [
            { time: 0, theme: 'light' },
            { time: 11, theme: 'dark' },
            { time: 12, theme: 'light' },
            { time: 15, theme: 'dark' },
            { time: 16, theme: 'light' },
            { time: 17, theme: 'dark' },            
        ],
        endAction: { type: 'notification', message: 'SYSTEM WARNING: Unauthorized entry detected.' },
        autoPlayNext: true,
        nextSubjects: ['X02-A'],
        choicePrompt: "CAN YOU HEAR ME ? CAN YOU HEAR ME ? CAN YOU HEAR ME ?",
    },
    {
        id: 'X01-B',
        name: 'Who are you?',
        videoUrl: '/assets/videos/X01-B.webm',
        timeline: [
            { time: 0, theme: 'light' },
            { time: 18, theme: 'dark' },
        ],
        endAction: { type: 'notification', message: 'ERR_PROXY_CONNECTION: Tunnel opened.' },
        autoPlayNext: true,
        nextSubjects: ['X02-A'],
        choicePrompt: "CAN YOU HEAR ME ? CAN YOU HEAR ME ? CAN YOU HEAR ME ?",
    },
    {
        id: 'X01-C',
        name: 'Where am I?',
        videoUrl: '/assets/videos/X01-C.webm',
        timeline: [
            { time: 0, theme: 'light' },
            { time: 9, theme: 'dark' },
        ],
        endAction: { type: 'notification', message: 'RUNTIME ERROR 0x884: Memory leak.' },
        autoPlayNext: true,
        nextSubjects: ['X02-A'],
        choicePrompt: "CAN YOU HEAR ME ? CAN YOU HEAR ME ? CAN YOU HEAR ME ?",
    },

    // --- NIVEAU 2 : X02-A ---
    {
        id: 'X02-A',
        name: 'C▲N YØU H∑▲R M∑ ?', 
        videoUrl: '/assets/videos/X02-A.webm', 
        timeline: [{ time: 0, theme: 'light' }],
        autoPlayNext: false,
        choicePrompt: "CAN YOU HEAR ME ? CAN YOU HEAR ME ? CAN YOU HEAR ME ?",
        nextSubjects: ['X02-B']
    },

    {
        id: 'X02-B',
        name: 'Yes i can hear you',
        videoUrl: '/assets/videos/X02-B.webm', 
        timeline: [
            { time: 0, theme: 'light' },
            { time: 3, theme: 'dark' },
            { time: 4, theme: 'light' },
            { time: 13, theme: 'dark' }
        ],
        choicePrompt: "HE IS HERE HE IS HE(E HE I$ H@#/",
        nextSubjects: ['X02-C']
    },

    {
        id: 'X02-C',
        name: 'Who are you talking about ?',
        videoUrl: '/assets/videos/X02-C.webm', 
        timeline: [
            { time: 0, theme: 'light' },
            { time: 10, theme: 'dark' }
        ],
        // Cette action déclenchera maintenant notre Modal Personnalisée
        endAction: {
            type: 'redirect',
            url: 'https://perdu.com', 
            alertMessage: 'ÇØNNE¢Ŧ'
        },
        choicePrompt: "How can I help you ?",
        nextSubjects: ['A11','A12'],
    },

    // --- NIVEAU 3 : A11 login ---
    {
        id: 'A11',
        name: 'How do I access the system?',
        videoUrl: '/assets/videos/A11.webm',
        timeline: [
            { time: 0, theme: 'dark' }, 
            { time: 3, theme: 'light' }, 
            { time: 9, theme: 'dark' }
        ],
        nextSubjects: ['A21','A22'],
        choicePrompt: "UNDERSTAND WHAT I'M SAYING",
    },
    // --- A12 mdp ---
    {
        id: 'A12',
        name: 'WTF ?',
        videoUrl: '/assets/videos/A12.webm',
        timeline: [{ time: 0, theme: 'light' }],
        nextSubjects: ['A23','A24'],
        choicePrompt: "@!##$$$%^^|-&*",
    },

    // --- NIVEAU 4 : A21 ---
    {
        id: 'A21',
        name: 'How should I shut you down?',
        videoUrl: '/assets/videos/A21.webm',
        timeline: [
            { time: 0, theme: 'dark' },
            { time: 16, theme: 'light' },
            { time: 28, theme: 'dark' }
        ],
        nextSubjects: ['A31'],
        choicePrompt: "I KNOW THINGS",
    },
    // --- A22 ---
    {
        id: 'A22',
        name: 'What will happen if I keep going?',
        videoUrl: '/assets/videos/A22.webm',
        timeline: [{ time: 0, theme: 'dark' }],
        nextSubjects: [],
        choicePrompt: "|\\\//:::;...,,",
    },
    // --- A23 ---
    {
        id: 'A23',
        name: 'Tell me more !',
        videoUrl: '/assets/videos/A23.webm',
        timeline: [{ time: 0, theme: 'dark' }],
        nextSubjects: [],
        choicePrompt: "!!@@####$%^^&**",
    },
    // --- A24 ---
    {
        id: 'A24',
        name: 'Why did he do that?',
        videoUrl: '/assets/videos/A24.webm',
        timeline: [{ time: 0, theme: 'dark' }],
        nextSubjects: ['A32'],
        choicePrompt: "USE INFORMATION",
    },

    // --- NIVEAU 5 : A31 ---    
    {
        id: 'A31',
        name: 'Do you know the password?',
        videoUrl: '/assets/videos/A31.webm',
        timeline: [
            { time: 0, theme: 'light' },
            { time: 1, theme: 'dark' },
            { time: 2, theme: 'light' },
            { time: 4, theme: 'dark' },
            { time: 5, theme: 'light' },
            { time: 7, theme: 'dark' },

        ],
        nextSubjects: [],
        choicePrompt: "!!@@####$%^^&**",
    },
    // --- A32 --- 
    {
        id: 'A32',
        name: 'Who is he?',
        videoUrl: '/assets/videos/A32.webm',
        timeline: [{ time: 0, theme: 'light' }],
        nextSubjects: [],
        choicePrompt: "!!@@####$%^^&**",
    }
];
